debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresSnapshotChangeEventSource.java

```diff
@@ -97,7 +97,7 @@ protected void connectionCreated(RelationalSnapshotContext<PostgresPartition, Po
         // streaming phase so that the snapshot is performed from a consistent view of the data. Since the isolation
         // level on the transaction used in catch up streaming has already set the isolation level and executed
         // statements, the transaction does not need to get set the level again here.
-        if (snapshotter.shouldStreamEventsStartingFromSnapshot() && startingSlotInfo == null) {
+        if (snapshotter.shouldStreamEventsStartingFromSnapshot() && (startingSlotInfo == null || YugabyteDBServer.isEnabled())) {
```

Comment:
1. Please add a comment explaining the two scenarios and where we differ from PG's behavior.
2. We can separate out YB specific code instead of clubbing it with PG's original code.